cmd/initContainer/main.go

```diff
@@ -445,42 +447,51 @@ func convertGR(pclient kyvernoclient.Interface) error {
 	}
 	for _, gr := range grs.Items {
 		cp := gr.DeepCopy()
-		ur := &kyvernov1beta1.UpdateRequest{
-			ObjectMeta: metav1.ObjectMeta{
-				GenerateName: "ur-",
-				Namespace:    config.KyvernoNamespace(),
-				Labels:       cp.GetLabels(),
-			},
-			Spec: kyvernov1beta1.UpdateRequestSpec{
-				Type:     kyvernov1beta1.Generate,
-				Policy:   cp.Spec.Policy,
-				Resource: cp.Spec.Resource,
-				Context: kyvernov1beta1.UpdateRequestSpecContext{
-					UserRequestInfo: kyvernov1beta1.RequestInfo{
-						Roles:             cp.Spec.Context.UserRequestInfo.Roles,
-						ClusterRoles:      cp.Spec.Context.UserRequestInfo.ClusterRoles,
-						AdmissionUserInfo: cp.Spec.Context.UserRequestInfo.AdmissionUserInfo,
-					},
-					AdmissionRequestInfo: kyvernov1beta1.AdmissionRequestInfoObject{
-						AdmissionRequest: cp.Spec.Context.AdmissionRequestInfo.AdmissionRequest,
-						Operation:        cp.Spec.Context.AdmissionRequestInfo.Operation,
+		var request *admissionv1.AdmissionRequest
+		if cp.Spec.Context.AdmissionRequestInfo.AdmissionRequest != "" {
+			var r admissionv1.AdmissionRequest
+			if err := json.Unmarshal([]byte(cp.Spec.Context.AdmissionRequestInfo.AdmissionRequest), &r); err == nil {
+				request = &r
+			}
+		}
+		if err == nil {
```

Comment:
What is this err?

Reply:
if unmashaling from the GR fails

Reply:
Isn't it checked on line 453?

Reply:
?

Reply:
I’ll check later, not in front of a computer right now.

Reply:
> I’ll check later, not in front of a computer right now. 

Sure.

Reply:
I have pushed the commit to fix the error handling.